python/spglib/spglib.py

```diff
@@ -97,6 +106,76 @@ class SpglibError:
 spglib_error = SpglibError()
 
 
+@dataclasses.dataclass(eq=True, frozen=True)
+class DictInterface(Mapping[str, "Any"], ABC):
+    """Base class for dataclass with dict interface."""
```

Comment:
```suggestion
@dataclasses.dataclass(eq=True, frozen=True)
class DictInterface(Mapping[str, "Any"], ABC):
    """
    Base class for dataclass with dict interface.
    
    .. versionadded:: 2.5.0
    .. deprecated:: 2.5.0
        Dict-like interface (``obj['field']``) are deprecated. Please use attribute interface instead (``obj.field``)
    """
```
the docstring might exceed 80 characters. Was also thinking of including `versionremoved`, but let's make a proper review of all of them and make a milestone for `Spglib 3.0`. I think we still have a lot to clean up before we could advertise these.

Also `.. versionadded:: 2.5.0` to the other classes and `get_magnetic_spacegroup_type_from_symmetry`